CONTRIBUTING.md

```diff
@@ -3,6 +3,61 @@
 We'd love to accept your patches and contributions to this project. There are
 just a few small guidelines you need to follow.
 
+## Contribution Guidelines
+
+To propose a new feature or a change that alters some existing user experience
+or creates a new user experience, follow these steps:
+
+### Step 1: Establish Context
+
+Search on KFP GitHub issues list to see if the same or similar proposal has been
+made in the past.  The historical context can help you draft a better
+proposal. Sometimes you will find a very similar proposal was already presented,
+discussed thoroughly, and that it is either awaiting contribution (in active
+development) or was rejected (often due to timing or conflicting scope with
+other plans). To avoid confusion and conflicts, where possible, please
+contribute to existing issues before creating new ones.
+
+### Step 2: Create Feature Request
+
+Create a new issue using the “Feature Request” template if no existing issue is
+found. Fill in answers to the template questions. To avoid delays, provide as
+much information as needed for initial review. Keep in mind that new features
+should comply with backward-compatibility and platform-portability requirements.
+
+### Step 3: Initial Team Triage
+
+Wait for a member from the [Kubeflow Pipelines
+team](https://github.com/kubeflow/internal-acls/blob/35a5f857723f72341a8841311f6f435b4833e457/github-orgs/kubeflow/org.yaml#L912-L931)
```

Comment:
nit: Not sure I have a strong feeling on this, but maybe we should just use the file path and the section of the YAML that contains it the members (`.orgs.kubeflow.pipelines.members`) instead of the canonical/commit URL, which will inevitably go out of date at some point.